src/transformers/models/time_series_transformer/configuration_time_series_transformer.py

```diff
@@ -135,7 +136,7 @@ class TimeSeriesTransformerConfig(PretrainedConfig):
 
     def __init__(
         self,
-        prediction_length: Optional[int] = None,
+        prediction_length: int = 24,
```

Comment:
@sgugger: @kashif originally thought a few arguments of this config's `__init__` should be positional arguments. 

- It seems we can't make positional arugments technically, as `to_json_string` will call `to_diff_dict` then `self.__class__().to_dict()` (so we always need to make `config_class()` works without specifying any  arg)
- And we also want it works this way, so easier for uers to get started (?)
- We use the values from a pretrained model as the default values for arguments of a config class

Is my understanding above correct?

As for `TimeSeriesTransformerModel`, there is no checkpoint, it's just @kashif trains a model on a tourism dataset quickly (3-5 minutes) and uploaded. And we just set values like `prediction_length (`int`, defaults to 24):` according to that one.
He says `24` has no real meaning, and the plan is to train models on all the time series datasets.

But I think it is fine (and we have to) put some default values anyway, and users can change them for their own datasets.

Reply:
Why not leave it as `None` then?

Yes we can't have config arguments that are positional, since the Transformers library is a library of pretrained model. So every argument of the config has a default corresponding to a canonical checkpoint associated to that model. This is quite an exceptional situation here for time series, as normally a model with no canonical checkpoint is not even accepted in the library ;-)

Leaving the argument as None is probably the way to go, and you can then assert in the config code if you want to force the user to always pass it.

Reply:
OK, thank you for spending time explain to me ❤️

Reply:
Not a problem!